lib/pbench/server/api/resources/query_apis/index_mappings.py

```diff
@@ -58,12 +58,17 @@ class IndexMappings(Resource):
     def __init__(self, logger: Logger):
         self.logger = logger
 
-    def get(self) -> Dict[AnyStr, Any]:
+    def get(self, index_name: AnyStr) -> Dict[AnyStr, Any]:
         """
-        Return mapping properties of the run-data document by querying the template database
+        Return mapping properties of the document specified by the index name in the URI.
+        For example, user can get the run document properties by making a GET request on
+        index/mappings/run. Similarly other index documents can be fetched by making a GET
+        request on appropriate index names.
+        We fetch the mapping by querying the template database. If the template is not found
+        in the database NOT_FOUND error will be raised.
         """
         try:
-            template = Template.find("run")
+            template = Template.find(index_name)
```

Comment:
GitHub isn't letting me comment on line 84, so I'll put it here. Now that `index_name` is a parameter, your log message should include the actual index name we couldn't find, not say `Run document template`.

(I think the `abort` is OK as is, though; it's reasonable to expect them to understand that if they asked for `/index/mappings/foobar` then it means `foobar` wasn't an index template name.)

Reply:
Fixed